sdk/javascript/test/test/sharedKeyTests.js

```diff
@@ -0,0 +1,79 @@
+const { PrivateKey, PublicKey } = require('../../src/CryptoTypes');
+const { expect } = require('chai');
+
+const runBasicSharedKeyTests = testDescriptor => {
+	// region mutual shared
+
+	const assertDerivedSharedResult = (mutate, assertion) => {
+		// Arrange:
+		const privateKey1 = PrivateKey.random();
+		const otherPublicKey1 = new testDescriptor.KeyPair(PrivateKey.random()).publicKey;
+
+		const privateKey2Bytes = new Uint8Array(privateKey1.bytes);
+		const otherPublicKey2Bytes = new Uint8Array(otherPublicKey1.bytes);
+
+		mutate(privateKey2Bytes, otherPublicKey2Bytes);
+
+		const keyPair1 = new testDescriptor.KeyPair(privateKey1);
+		const keyPair2 = new testDescriptor.KeyPair(new PrivateKey(privateKey2Bytes));
+
+		// Act:
+		const sharedKey1 = testDescriptor.deriveSharedKey(keyPair1, otherPublicKey1);
+		const sharedKey2 = testDescriptor.deriveSharedKey(keyPair2, new PublicKey(otherPublicKey2Bytes));
+
+		// Assert:
+		assertion(sharedKey1, sharedKey2);
+	};
+
+	it('shared keys generated with same intpus are equal', () => {
+		assertDerivedSharedResult(() => {}, (lhs, rhs) => expect(lhs).to.deep.equal(rhs));
+	});
+
+	it('shared keys generated for different private keys are different', () => {
+		assertDerivedSharedResult(privateKeyBytes => { privateKeyBytes[0] ^= 0xFF; }, (lhs, rhs) => expect(lhs).to.not.deep.equal(rhs));
+	});
+
+	it('shared keys generated for different other public keys are different', () => {
+		for (let i = 1; 256 > i; ++i) {
+			try {
+				const mutate = (_, otherPublicKey) => { otherPublicKey[0] ^= i; };
+				assertDerivedSharedResult(mutate, (lhs, rhs) => expect(lhs).to.not.deep.equal(rhs));
+				break;
+			} catch (error) {
+				// comments for eslint purposes, no need to handle this
+			}
+		}
+	});
+
+	it('mutual shared results are equal', () => {
+		// Arrange:
+		const keyPair1 = new testDescriptor.KeyPair(PrivateKey.random());
+		const keyPair2 = new testDescriptor.KeyPair(PrivateKey.random());
+
+		// Act:
+		const sharedKey1 = testDescriptor.deriveSharedKey(keyPair1, keyPair2.publicKey);
+		const sharedKey2 = testDescriptor.deriveSharedKey(keyPair2, keyPair1.publicKey);
+
+		// Assert:
+		expect(sharedKey1).to.deep.equal(sharedKey2);
+	});
+
+	// endregion
+
+	// region invalid public key
+
+	it('public key not on the curve throws', () => {
```

Comment:
I think this test my occasionally fail : read point might end up on curve, and it _seems_ it might be failing more often then py variant, which in turn suggest that server-based isCanonical check might be bit too permissive...
will need to investigate

Reply:
explained on slack, the issue was kind of PEBKAC, try catch around expect.to.throw was a bad idea, similar issue was in py, had to replace both with some manual checks